packages/ui/components/forms/Input/index.js

```diff
@@ -12,94 +12,97 @@ import Select from '../Select'
 import TextInput from '../TextInput'
 import themed from '../../../theming/themed'
 
-const INPUTS = {
-  text: {
-    Component: TextInput,
-    getProps: $value => ({
-      value: $value && $value.get(),
-      // TODO: Use stringInsert and stringRemove
-      onChangeText: value => $value && $value.setDiff(value)
-    })
-  },
-  checkbox: {
-    Component: Checkbox,
-    getProps: $value => ({
-      value: $value && $value.get(),
-      onChange: value => $value && $value.setDiff(value)
-    })
-  },
-  object: {
-    Component: ObjectInput,
-    getProps: $value => ({
-      value: $value && $value.get()
-    })
-  },
-  array: {
-    Component: ArrayInput,
-    getProps: $value => ({
-      value: $value && $value.get()
-    })
-  },
-  number: {
-    Component: NumberInput,
-    getProps: $value => ({
-      value: $value && $value.get(),
-      onChangeNumber: value => $value && $value.setDiff(value)
-    })
-  },
-  select: {
-    Component: Select,
-    getProps: $value => ({
-      value: $value && $value.get(),
-      onChange: value => $value && $value.setDiff(value)
-    })
-  },
-  date: {
-    Component: DateTimePicker,
-    getProps: $value => ({
-      date: $value && $value.get(),
-      onDateChange: value => $value && $value.setDiff(value),
-      mode: 'date'
-    })
-  },
-  datetime: {
-    Component: DateTimePicker,
-    getProps: $value => ({
-      date: $value && $value.get(),
-      onDateChange: value => $value && $value.setDiff(value),
-      mode: 'datetime'
-    })
-  },
-  time: {
-    Component: DateTimePicker,
-    getProps: $value => ({
-      date: $value && $value.get(),
-      onDateChange: value => $value && $value.setDiff(value),
-      mode: 'time'
-    })
-  },
-  password: {
-    Component: PasswordInput,
-    getProps: $value => ({
-      value: $value && $value.get(),
-      onChangeText: value => $value && $value.setDiff(value)
-    })
-  }
-}
-const INPUT_TYPES = Object.keys(INPUTS)
-
 function Input ({
   style,
   error,
   type,
   $value,
   ...props
 }) {
-  if (!type || !INPUT_TYPES.includes(type)) {
+  const { inputs, types } = useMemo(() => {
+    // INFO: because ObjectInput uses Input inside
+    const _inputs = {
```

Comment:
we can't move this code outside of the component because ObjectInput uses Input inside which generates error `minified react error`